File: LeetCode/3477.Fruits_Into_Baskets_II/3477.Fruits_Into_Baskets_II.py
#!/usr/bin/env python3


class Solution:
    def numOfUnplacedFruits(self, fruits: List[int], baskets: List[int]) -> int:
        # leftmost capcity >= qty of fruits
        # each basket one type
        # return num of fruits type unplaced

        n = len(fruits)
        res = 0
        for i, f in enumerate(fruits):
            found = False
            for j in range(n):
                if f <= baskets[j]:
                    baskets[j] = 0
                    found = True
                    break
            if not found:
                res += 1

        return res

File: LeetCode/3477.Fruits_Into_Baskets_II/3477.Fruits_Into_Baskets_II.swift
// TC:O(n**2)
class Solution {
    func numOfUnplacedFruits(_ fruits: [Int], _ baskets: [Int]) -> Int {
        // return type that remain unplaced
        let n = fruits.count
        var baskets = baskets

        for f in fruits {
            for (i, b) in baskets.enumerated() {
                if b >= f {
                    baskets[i] = 0
                    break
                }
            }
        }

        return baskets.filter { $0 > 0 }.count
    }
}

// segment tree
